README.md

```diff
@@ -1 +1,69 @@
-# BlockNLPAlgorithms.jl
\ No newline at end of file
+# BlockNLPAlgorithms.jl
+This package implements several popular decomposition algorithms for block-structured nonlinear optimization problems (NLPs). This package requires the NLPs to be implemented as `AbstractBlockNLPModels`; for more details on modeling block-structured NLPs, please refer to the documentation of [BlockNLPModels.jl](https://github.com/exanauts/BlockNLPModels.jl).
+
+## Installation
+This package is currently under active development and has not been registered yet. However, to access the current code, one can enter the following command in Julia REPL:
+
+```julia
+]add "https://github.com/exanauts/BlockNLPAlgorithms.jl"
+```
+
+To confirm if the package was installed correctly, please run:
+```julia
+test BlockNLPAlgorithms
+```
+The test code generates and solves a small instance of a `BlockNLPModel` with all the implemented algorithms using [MadNLP.jl](https://github.com/MadNLP/MadNLP.jl) as the subproblem solver.
+
+## Implemented Decomposition Algorithms
+1. [Dual Decomposition Algorithm](http://www.seas.ucla.edu/~vandenbe/236C/lectures/dualdecomp.pdf)
+2. [Alternating Direction Method of Multipliers or ADMM](https://stanford.edu/~boyd/admm.html)
+3. [Proximal ADMM](http://alpha.math.uga.edu/~mjlai/papers/DLPYfinal.pdf)
+
+## Quickstart Guide
+To solve a `AbstractBlockNLPModel` with a decomposition algorithms implemented in this package, one can use the following method:
+
+`solve(model::AbstractBlockNLPModel,method::Type{<:AbstractBlockNLPSolver};options...)`, where
+
+- `model` is the block nlp
+- `method` refers to the choice of the decomposition algorithms, user can choose one of the three available options:
+    - BlockNLPAlgorithms.DualDecomposition
+    - BlockNLPAlgorithms.ADMM
+    - BlockNLPAlgorithms.ProxADMM
+- `options` are the optional arguments to customize the behavior of decomposition solver. Available solver options are listed in the next subsection.
```

Comment:
Suggestion:
`the behavior of decomposition solver` -> `the behavior of the decomposition solver`